### ISSUE TITLE
Unable to login when app is accessed via `vaadinServlet`

### ISSUE THREAD

**Issue by fluorumlabs** [vaadin/flow#11001]
Issue manifestation is similar to https://github.com/vaadin/flow/issues/10997 but also resembles https://github.com/vaadin/flow/issues/11000. The proposed fix (https://github.com/vaadin/flow/pull/10999/files) doesn't change anything.

Steps to reproduce:
1. Create SVC project
2. Follow https://vaadin.com/docs/latest/fusion/security/spring-login docs
3. Start the application and navigate to http://localhost:8080/vaadinServlet/login
4. Login

Expected behaviour:

- You are logged in

Actual behaviour:

- You are not logged in; clicking "login" makes a call to http://localhost:8080/vaadinServlet/login which returns 200 with a HTML content.

**Comment by Legioth** [comment]
I would say that the expected behaviour is that http://localhost:8080/vaadinServlet/login wouldn't work at all, i.e. that the internally used `vaadinServlet` mapping wouldn't be publicly accessible at all.

**Comment by Artur-** [comment]
Yes, the bug is that `/vaadinServlet/*` exists and should be handled by
1. Removing the need for it to be publicly available (push seems to use it)
2. Removing it from the Spring Secuity conf https://github.com/vaadin/spring/issues/789
3. Make it not available
